azurelinuxagent/common/conf.py

```diff
@@ -547,6 +548,19 @@ def get_auto_update_to_latest_version(conf=__conf__):
     return conf.get_switch("AutoUpdate.UpdateToLatestVersion", default)
 
 
+def get_protocol_endpoint_discovery(conf=__conf__):
+    return conf.get("Protocol.EndpointDiscovery", "dhcp")
```

Comment:
This option didn't seem to fit in any existing (Logs, OS, Extensions, Provisioning, ResourceDisk, AutoUpdate).
I created Protocol.* option but open to suggestions

Reply:
LGTM